nodewatcher/modules/monitor/http/survey/processors.py

```diff
@@ -0,0 +1,80 @@
+from django.utils.translation import gettext_noop as _
+
+from django_datastream import datastream
+
+from nodewatcher.core.monitor import processors as monitor_processors
+from nodewatcher.modules.monitor.sources.http import processors as http_processors
+from nodewatcher.modules.monitor.datastream import fields as ds_fields, models as ds_models
+from nodewatcher.modules.monitor.datastream.pool import pool as ds_pool
+
+
+class SurveyInfoStreams(ds_models.RegistryRootStreams):
+    neighbor_graph = ds_fields.GraphField(tags={
+        'title': _("Neighbor topology"),
+        'description': _("Neighbor topology."),
+    })
+
+    def get_module_name(self):
+        return 'monitor.http.survey'
+
+    def get_stream_highest_granularity(self):
+        return datastream.Granularity.Hours
+
+
+class SurveyInfoStreamsData(object):
+    def __init__(self, node, vertices, edges):
+        self.node = node
+        self.neighbor_graph = {
+            'v': vertices,
+            'e': edges
+        }
+
+
+ds_pool.register(SurveyInfoStreamsData, SurveyInfoStreams)
+
+
+class SurveyInfo(monitor_processors.NodeProcessor):
+    """
+    Stores neighbor's reported channel, BSSID, SSID and signal strength into the datastream. A star shaped graph
+    is constructed. The source vertex contains an array of all BSSIDs of that node. Source vertex is stored according
+    to its UUID, wherease all neighbors are stored according to their BSSIDs. Will only run if HTTP monitor module
+    has previously fetched data.
+    """
+
+    @monitor_processors.depends_on_context("http", http_processors.HTTPTelemetryContext)
+    def process(self, context, node):
+        """
+        Called for every processed node.
+
+        :param context: Current context
+        :param node: Node that is being processed
+        :return: A (possibly) modified context
+        """
+
+        version = context.http.get_module_version('core.wireless')
+        if version == 0:
+            # Unsupported version or data fetch failed
+            return context
+        edges, vertices = [], []
+        try:
+            vertex_bssids = []
+            for interface in context.http.core.wireless.interfaces.values():
+                vertex_bssids.append(interface['bssid'])
+            vertex = dict(i=str(node.uuid), bssids=vertex_bssids)
+            vertices.append(vertex)
+            for radio in context.http.core.wireless.radios.values():
+                for neighbor in radio.survey:
+                    vertices.append(dict(i=neighbor['bssid']))
+                    # Add an edge for that neighbor
+                    edge = {'f': str(node.uuid), 't': neighbor['bssid']}
+                    edge['channel'] = neighbor['channel']
```

Comment:
Oh, `survey` is an array, so the returned things are simple dicts and not our context objects. @kostko, you agree?

Then leave it as it is. But be careful, if there is no `channel` in the `neighbor` dict, for example, this will throw an exception `KeyError`. You could maybe do:

```
if 'channel' in neighbor:
  edge['channel'] = neighbor['channel']
```

Or you can skip the whole edge, if you want. This is on you to decide how you want to deal with invalid/incomplete data. Currently, `KeyError` is then catched and the whole parsing stops, and only data parsed until then is stored. This is a bit strange. I would say we should or continue parsing other iterations, or do not store anything. Not partial data. @kostko?